engine.py

```diff
@@ -55,22 +55,42 @@ def fight(self):
         After attack/support it changes the team to another one.
         :return: Stops attack if chosen character is stunned
         """
+        # # working version
+        # while all(game.teams_list):
+        #     self.rounds += 1  # counts the rounds
+        #     char1 = self.choose_attacking_character()
+        #
+        #     for modi in char1.modifier_list:
+        #         modi.act()
+        #
+        #     char1.get_exp()
+        #     if type(char1) is Support:  # checks if active character is a support if yes he cast spell on random
+        #         # character from its team
+        #         char2 = self.choose_attacking_character()
+        #         char1.next_move(char2)
+        #         self.change_team_order()
+        #     else:  # use act typical for its character on the random opponent character
+        #         self.change_team_order()
+        #         char2 = self.choose_attacking_character()
+        #         char1.next_move(char2)
+
         while all(game.teams_list):
             self.rounds += 1  # counts the rounds
             char1 = self.choose_attacking_character()
+
             for modi in char1.modifier_list:
                 modi.act()
 
             char1.get_exp()
             if type(char1) is Support:  # checks if active character is a support if yes he cast spell on random
                 # character from its team
                 char2 = self.choose_attacking_character()
-                char1.next_move(char2)
+                char1.next_move(Team.find_weakest_opponent(char2.team))
```

Comment:
Nie tak się to wywołuje.
Nie odołuj się do metod niestatycznych w ten sposób (przez klasę). Wybierz team i niech on sprawdzi kto u niego jest najsłabszy.